lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1859,6 +1864,47 @@ ChangeResult TypeAnalyzer::visitAtenAddCLikeOp(
   return getLatticeElement(op->getResult(0)).join(knowledge);
 }
 
+ChangeResult TypeAnalyzer::visitAtenNativeBatchNormOp(
+    AtenNativeBatchNormOp op,
+    ArrayRef<LatticeElement<ValueKnowledge> *> operands) {
+  auto input = operands[0]->getValue();
+
+  auto batchNormKnowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+  auto meanKnowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+  auto invStdKnowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+
+  batchNormKnowledge.dtype = input.dtype;
+  meanKnowledge.dtype = input.dtype;
+  invStdKnowledge.dtype = input.dtype;
+
+  // Rank of the input tensor must be greater than or equal to 2. The size of
+  // the input tensor as well as the output tensor should be (N, C, D?, H?, W?).
+  // The running_mean, running_var, weight, and bias should be of size (C).
+  bool training = false;
+  if (matchPattern(op.training(), m_TorchConstantBool(&training)) &&
+      input.hasSizes && input.sizes.size() >= 2) {
+    batchNormKnowledge.hasSizes = true;
+    meanKnowledge.hasSizes = true;
+    invStdKnowledge.hasSizes = true;
+    batchNormKnowledge.sizes = input.sizes;
+    meanKnowledge.sizes.resize(1, 0);
```

Comment:
Is the return size rank 0 or rank 1 with size 0? if it's rank 0, I think we don't need to `resize(1, 0)`

Reply:
From running one of the unit tests, it seems that the result is rank 1 with size 0

```python
In [13]: _, mean, var = torch.ops.aten.native_batch_norm(torch.rand(2,5,2,3), torch.rand(5), torch.rand(5), 
                                                         torch.rand(5), torch.rand(5), training=False, momentum=0.1, eps=0.0001)

In [14]: mean.shape
Out[14]: torch.Size([0])

In [15]: var.shape
Out[15]: torch.Size([0])
```

Reply:
Yes, it is rank 1 size 0. :)